src/+energy/calculateEnergyParameters.m

```diff
@@ -61,23 +73,30 @@
         % terms.(0.0003,volumetric heat capacity)
         if ~ModelSettings.Soilairefc
             QL(i) = -(KLhBAR(i) * DhDZ(i) + (KLTBAR(i) + DTDBAR(i)) * DTDZ(i) + KLhBAR(i));
+            QL_h(i) = -(KLhBAR(i) * DhDZ(i) + KLhBAR(i)); % added by Mostafa
+            QL_T(i) = -((KLTBAR(i) + DTDBAR(i)) * DTDZ(i)); % added by Mostafa
+            QL_a(i) = 0; % added by Mostafa
             Qa = 0;
         else
             Qa = -((DEhBAR + GasDispersivity.D_Vg(i)) * DRHODAz(i) - RHODA(i) * (GasDispersivity.V_A(i) + Constants.Hc * QL(i) / Constants.RHOL));
         end
 
         if SoilVariables.DVa_Switch == 1
             QV = -(DEhBAR + GasDispersivity.D_Vg(i)) * DRHOVhDz(i) * DDhDZ(i) - (DEhBAR * EtaBAR + GasDispersivity.D_Vg(i)) * DRHOVTDz(i) * DTDZ(i) + RHOVBAR * GasDispersivity.V_A(i);
+            QVH(i) = -(DEhBAR + GasDispersivity.D_Vg(i)) * DRHOVhDz(i) * DDhDZ(i); % activated by Mostafa
+            QVT(i) = -(DEhBAR * EtaBAR + GasDispersivity.D_Vg(i)) * DRHOVTDz(i) * DTDZ(i); % activated by Mostafa
+            QVa(i) = RHOVBAR * GasDispersivity.V_A(i); % added by Mostafa
         else
             QV = -(DEhBAR + GasDispersivity.D_Vg(i)) * DRHOVhDz(i) * DDhDZ(i) - (DEhBAR * EtaBAR + GasDispersivity.D_Vg(i)) * DRHOVTDz(i) * DTDZ(i);
+            QVH(i) = -(DEhBAR + GasDispersivity.D_Vg(i)) * DRHOVhDz(i) * DDhDZ(i); % activated by Mostafa
+            QVT(i) = -(DEhBAR * EtaBAR + GasDispersivity.D_Vg(i)) * DRHOVTDz(i) * DTDZ(i); % activated by Mostafa
+            QVa(i) = 0; % added by Mostafa
```

Comment:
Three issues here:

1. The equations of `QVH` and `QVT` are exactly the same before and after `else`. is this correct? if so, they can be moved before the if statement `if SoilVariables.DVa_Switch == 1`
2. can you add a reference for the new equations? 
3. remove duplications of comments `% activated by Mostafa` and `% added by Mostafa`

Reply:
> Three issues here:
> 
> 1. The equations of `QVH` and `QVT` are exactly the same before and after `else`. is this correct? if so, they can be moved before the if statement `if SoilVariables.DVa_Switch == 1`
Yes, no change. I moved them before the if statement `if SoilVariables.DVa_Switch == 1`

> 2. can you add a reference for the new equations?
No reference, the new equations were already there before but were commented so I activated them.

> 3. remove duplications of comments `% activated by Mostafa` and `% added by Mostafa`
Done